File: java/logica_y_validaciones/validar_correo_or_telefono.java
package logica_y_validaciones;
public class validar_correo_or_telefono {
    public static void main(String[] args) {
        String correo = "usuario@gmail.com";
        String telefono = "";
        boolean correoValido = correo.contains("@") && correo.contains(".");
        boolean telefonoValido = telefono.length() == 10 && telefono.matches("\\d+");
        if (correoValido || telefonoValido) {
            System.out.println("Contacto válido.");
        } else {
            System.out.println("Debe ingresar un correo o un teléfono válido.");
        }
    }
}

File: visual_basic/operaciones_logicos/validar_correo_or_telefono/validar_correo_or_telefono .vb
Imports System
Module validar_correo_or_telefono
    Sub Main(args As String())
        Dim correo As String
        Dim telefono As String
        Console.Write("Ingrese el correo: ")
        correo = Console.ReadLine()
        Console.Write("Ingrese el teléfono: ")
        telefono = Console.ReadLine()
        If correo.Trim() <> "" Or telefono.Trim() <> "" Then
            Console.WriteLine("Dato válido ✅ (al menos uno fue ingresado)")
        Else
            Console.WriteLine("Error: Debe ingresar correo o teléfono ❌")
        End If
    End Sub
End Module